### PR TITLE
Run vscode tests in the CI

### PR DESCRIPTION
**Pull request by rrousselGit**

### CODE REVIEW

File: .github/workflows/node-test.yml [hunk 1]
```diff
@@ -38,6 +38,34 @@ jobs:
       - run: npm i -g npm@9.5
       - run: npm ci
       - run: npm run lint:changed-files
+      - run: npm run lint
+        working-directory: firebase-vscode
+
+  vscode_unit:
+    runs-on: windows-latest
```
Comment (rrousselGit) [inline .github/workflows/node-test.yml:45]:
```suggestion
    # Using windows to bypass an error thrown by VScode
    # when run in an environment that does not have a screen.
    runs-on: windows-latest
```

**Review by hlshen — APPROVED** [review]
approved, going to let Joe take a look at this too

File: .github/workflows/node-test.yml [hunk 2]
```diff
@@ -38,6 +38,36 @@ jobs:
...
+        node-version:
+          - "18"
+          - "20"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: npm
+          cache-dependency-path: npm-shrinkwrap.json
```
Comment (joehan) [inline .github/workflows/node-test.yml:59]: Should this be firebase-vscode/[package-lock.json instead (or in addition)? I assume we want to cache the vscode deps here.

File: .github/workflows/node-test.yml [hunk 3]
```diff
@@ -38,6 +38,36 @@ jobs:
...
+
+      - run: npm i -g npm@9.5
+      - run: npm ci
```
Comment (joehan) [inline .github/workflows/node-test.yml:62]: Double checking - if you remove this line, do the tests still pass?

**Review by joehan — APPROVED** [review]
Some questions/potential minor efficiency improvements, but very glad to be adding these to CI.

File: .github/workflows/node-test.yml (hunk 3 above)
Comment (rrousselGit) [inline .github/workflows/node-test.yml:62]:
```suggestion
```

File: .github/workflows/node-test.yml [hunk 4]
```diff
@@ -38,6 +38,35 @@ jobs:
...
+        node-version:
+          - "18"
+          - "20"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: npm
+          cache-dependency-path: npm-shrinkwrap.json
```
Comment (rrousselGit) [inline .github/workflows/node-test.yml:59]:
```suggestion
          cache-dependency-path: firebase-vscode/package-lock.json
```

File: .github/workflows/node-test.yml (hunk 3 above)
Comment (rrousselGit) [reply .github/workflows/node-test.yml]: I doubt it. We have to run `npm i` in both the root and the vscode folder.
The `npm i -g` above doesn't count as it only updates the npm version and nothing else.

File: .github/workflows/node-test.yml [hunk 5]
```diff
@@ -38,6 +38,35 @@ jobs:
...
+
+      - run: npm i -g npm@9.5
```
Comment (rrousselGit) [inline .github/workflows/node-test.yml:61]:
```suggestion
      - run: npm i -g npm@9.5
      - run: npm ci
```

File: .github/workflows/node-test.yml [hunk 6]
```diff
@@ -38,6 +38,36 @@ jobs:
...
+      - run: npm ci
+
```
Comment (rrousselGit) [inline .github/workflows/node-test.yml:63]:
```suggestion
```

**Pull request merged**